src/west/config.py

```diff
@@ -58,37 +80,65 @@ def read_config():
     '''
 
     # Gather (potential) configuration file paths
+    files = []
 
     # System-wide and user-specific
+    if config_file == ConfigFile.ALL or config_file == ConfigFile.SYSTEM:
+        files.append(ConfigFile.SYSTEM.value)
 
-    if platform.system() == 'Linux':
+    if config_file == ConfigFile.ALL and platform.system() == 'Linux':
         # Probably wouldn't hurt to check $XDG_CONFIG_HOME (defaults to
-        # ~/.config) on all systems. It's listed in git-config(1). People were
-        # iffy about it as of writing though.
-        files = ['/etc/westconfig',
-                 os.path.join(os.environ.get('XDG_CONFIG_HOME',
-                                             os.path.expanduser('~/.config')),
-                              'west', 'config')]
-
-    elif platform.system() == 'Darwin':  # Mac OS
-        # This was seen on a local machine ($(prefix) = /usr/local)
-        files = ['/usr/local/etc/westconfig']
-    elif platform.system() == 'Windows':
-        # Seen on a local machine
-        files = [os.path.expandvars('%PROGRAMDATA%\\west\\config')]
+        # ~/.config) on all systems. It's listed in git-config(1). People
+        # were iffy about it as of writing though.
+        files.append(os.path.join(os.environ.get(
+            'XDG_CONFIG_HOME',
+            os.path.expanduser('~/.config')),
+            'west', 'config'))
 
-    files.append(os.path.expanduser('~/.westconfig'))
+    if config_file == ConfigFile.ALL or config_file == ConfigFile.GLOBAL:
+        files.append(ConfigFile.GLOBAL.value)
 
     # Repository-specific
 
-    files.append(os.path.join(west_dir(), 'config'))
+    if config_file == ConfigFile.ALL or config_file == ConfigFile.LOCAL:
+        files.append(os.path.join(west_dir(), ConfigFile.LOCAL.value))
 
     #
     # Parse all existing configuration files
     #
     config.read(files, encoding='utf-8')
 
 
+def update_config(section, key, value, configfile=ConfigFile.LOCAL):
+    '''
+    Updates 'key' in section 'section' in ConfigFile, creating
+    'section' if it does not exist and write the file afterwards.
+
+    If value is None/empty, 'key' is left as-is.
+    '''
+    if not value:
+        return
+
+    if configfile == ConfigFile.ALL:
+        # Not possible to update ConfigFile.ALL, needs specific conf file here.
+        return
```

Comment:
doesn't really matter.
The `configfile` provided is is not coming from user, but from another `west` module so someone is doing something really wrong.
 But will change to `ValueError` as it might be more helpful if someone starts to update conf files elsewhere.

Reply:
fixed

Reply:
> But will change to ValueError as it might be more helpful if someone starts to update conf files elsewhere.

Exactly, I'm thinking about using this module from elsewhere, like extensions or other programs that use west. Especially when we merge the multirepo and config commands into what gets shipped via pypi, this will all be importable.